src/openms/source/KERNEL/MSChromatogram.cpp

```diff
@@ -415,3 +415,24 @@ void MSChromatogram::clear(bool clear_meta_data)
     integer_data_arrays_.clear();
   }
 }
+
+void MSChromatogram::mergePeaks(MSChromatogram& other)
+{
+  vector<ChromatogramPeak> temp;
+  std::merge(this->begin(), this->end(), other.begin(), other.end(), std::back_inserter(temp), ChromatogramPeak::RTLess());
```

Comment:
would be interesting to benchmark if appending the other vector and a sort is faster. @cbielow what do you think?

Reply:
I think I would take the merge code and instead of merging the element add it on top if they are equal in RT. that way you don't need the extra copying.

Reply:
or even closer: https://en.cppreference.com/w/cpp/algorithm/set_union
this merges two ranges but keeps only one element if two are equal. I did not check but this sounds like it could be adapted to create the summed up version?

Reply:
Yes, I think I can do something clever with a modified set_union interpretation. I'll let you know.

Reply:
Got it working.